util/updateXML/update1.8.0.6.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, RTE (http://www.rte-france.com)
+# See AUTHORS.txt
+# All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, you can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of Dynawo, an hybrid C++/Modelica open source suite
+# of simulation tools for power systems.
+
+from content.Ticket import ticket
+
+# Convert parameter RefFlag from a Boolean into an Integer
+
+@ticket(3583)
+def update(jobs):
+    weccs = jobs.dyds.get_bbms(lambda bbm: bbm.get_lib_name() == "BESSWeccCurrentSource" or bbm.get_lib_name() == "BESSWeccCurrentSourceNoPlantControl")
```

Comment:
To be clearer you could create 1 table with lib names 

`names = ["BESSWeccCurrentSource", "BESSWeccCurrentSourceNoPlantControl", "PhotovoltaicsWeccCurrentSourceB"] `

and another with the corresponding parameter name

`param_prefix= ["BESS", "BESS, "photovoltaics"...]`

With this only one loop is required
```
index = 0
for name in names:
   prefix = param_prefix [index]
    bbms = jobs.dyds.get_bbms(lambda bbm: bbm.get_lib_name() == name)
   for bbm in bbms:
        value = bbm.parset.get_param_value(prefix + "_RefFlag")
        bbm.parset.remove_param_or_ref(prefix + "_RefFlag")
        bbm.parset.add_param("INT", prefix + "_RefFlag", if value = true then 1 else 0)
  index += 1
```

Reply:
Thank you for the explanation.